File: leetcode/editor/cn/model/list_node.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# 类定义过程中使用注解引用类本身
from __future__ import annotations


class ListNode:
    def __init__(self, val=0, nextNode: ListNode = None):
        self.val = val
        self.next = nextNode

File: leetcode/editor/cn/model/ListNode.java
package model;

public class ListNode {

    public int val;
    public ListNode next;

    public ListNode() {
    }

    public static ListNode newList(int[] nums) {
        ListNode head = null;
        for (int i = nums.length - 1; i >= 0; i--) {
            head = new ListNode(nums[i], head);
        }
        return head;
    }

    public ListNode(int val) {
        this.val = val;
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }


}
